bot/services/ChatBot.cs

```diff
@@ -334,6 +365,12 @@ await foreach (var chatResponse in streamingCall.ResponseStream.ReadAllAsync(can
 
     protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
     {
+        // verify authorization
+        if (!this.IsAuthorized(turnContext))
+        {
+            throw new Exception("the activity was not authorized to use this bot.");

```

Comment:
suggest to say: the tenant was not authorized to use this bot

Reply:
The log does refer to the tenant, but outside the IsAuthorized method we don't know the reason it was denied. In theory there could be other checks than just tenant.